chainforge/react-server/src/ModelSettingSchemas.js

```diff
@@ -1000,6 +1000,79 @@ const AlephAlphaLuminousSettings = {
   },
 };
 
+const OllamaSettings = {
+  fullName: "Ollama",
+  schema: {
+    type: "object",
+    required: ["shortname"],
+    properties: {
+      shortname: {
+        type: "string",
+        title: "Nickname",
+        description:
+          "Unique identifier to appear in ChainForge. Keep it short.",
+        default: "Ollama",
+      },
+      ollamaModel: {
+        type: "string",
+        title: "Model",
+        description:
+          "Select a model to query using Ollama's API. Make sure you've pulled the model before. For more details, check out https://ollama.ai/library",
+        default: "mistral",
+      },
+      ollama_url: {
+        type: "string",
+        title: "URL",
+        description: "URL of the Ollama server generate endpoint.",
+        default: "http://localhost:11434/api/generate",
+      },
+      temperature: {
+        type: "number",
+        title: "temperature",
+        description: "Amount of randomness injected into the response. Ranges from 0 to 1. Use temp closer to 0 for analytical / multiple choice, and temp closer to 1 for creative and generative tasks.",
+        default: 1.0,
+        minimum: 0,
+        maximum: 1.0,
+        multipleOf: 0.01,
+      },
+      raw: {
+        type: "boolean",
+        title: "raw",
+        description: "Whether to disable the templating done by Ollama. If true, you'll need to insert annotations ([INST]) in your prompt.",
+        default: false,
+      },
+      top_k: {
+        type: "integer",
+        title: "top_k",
+        description: "Only sample from the top K options for each subsequent token. Used to remove \"long tail\" low probability responses. Defaults to -1, which disables it.",
+        minimum: -1,
+        default: -1
+      },
+      top_p: {
+        type: "number",
+        title: "top_p",
+        description: "Does nucleus sampling, in which we compute the cumulative distribution over all the options for each subsequent token in decreasing probability order and cut it off once it reaches a particular probability specified by top_p. Defaults to -1, which disables it. Note that you should either alter temperature or top_p, but not both.",
+        default: -1,
+        minimum: -1,
+        maximum: 1,
+        multipleOf: 0.001,
+      },
+      seed: {
+        type: "integer",
+        title: "seed",
+        description: "If specified, the OpenAI API will make a best effort to sample deterministically, such that repeated requests with the same seed and parameters should return the same result. Determinism is not guaranteed.",
+        allow_empty_str: true,
+      },
+      stop_sequences:  {
+        type: "string",
+        title: "stop_sequences",
+        description: "Sequences where the API will stop generating further tokens. Enclose stop sequences in double-quotes \"\" and use whitespace to separate them.",
+        default: ""
+      },
+    },
+  },
```

Comment:
I saw that other settings have `uiSchema` and `postprocessors` but I wasn't sure how those were used. Everything seems to be working without them so maybe they are not needed in this case?

Reply:
Hmm I'm not sure, usually I add uiSchema for nicer UI. `postprocessors` are not necessary unless there's a special type of parameter (like a variable-sized array of strings for stop sequences).